esmvalcore/preprocessor/_supplementary_vars.py

```diff
@@ -65,47 +78,62 @@ def add_cell_measure(cube, cell_measure_cube, measure):
         raise ValueError(
             f"measure name must be 'area' or 'volume', got {measure} instead"
         )
-    measure = iris.coords.CellMeasure(
-        cell_measure_cube.core_data(),
+    coord_dims = tuple(
+        range(cube.ndim - len(cell_measure_cube.shape), cube.ndim)
+    )
+    cell_measure_data = cell_measure_cube.core_data()
+    if cell_measure_cube.has_lazy_data():
+        cube_chunks = tuple(cube.lazy_data().chunks[d] for d in coord_dims)
+        cell_measure_data = cell_measure_data.rechunk(cube_chunks)
+    cell_measure = iris.coords.CellMeasure(
+        cell_measure_data,
```

Comment:
what if the data is not lazy? `cell_measure_data` will not be defined for that. EDIT: bugger, am blind - it's above as `core_data()` - but pls remind me, is `core_data()` defaults to simply realized `data` if the data is not lazy?

Reply:
fab, my memory is not yet failing me :grin: